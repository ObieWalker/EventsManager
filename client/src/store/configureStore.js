import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers/rootReducer';

const middleware = applyMiddleware(thunkMiddleware, createLogger({ collapsed: true }));

const Store = createStore(
    rootReducer,
    applyMiddleware(middleware)
);

export default Store;