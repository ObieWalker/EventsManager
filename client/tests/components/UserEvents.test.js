import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import connect, { UserEvents }
  from '../../src/components/UserDashboard/UserEvents';
import { events } from '../__mocks__/eventsData';
import { users } from '../__mocks__/usersData';

let mountedComponent;
let props;
const allUserEvents = {
  events: [events],
  moreHistory: true
};
const user = users.user2;
let moreEvents;
const deleteEvent = jest.fn();
const editEvent = jest.fn();
const getUsersEvents = jest.fn(() => Promise.resolve({}));

const mockStore = configureStore();
let wrapper;

const getComponent = () => {
  if (!mountedComponent) {
    props = {
      getUsersEvents,
      allUserEvents,
      user,
      moreEvents,
      events,
      deleteEvent,
      editEvent
    };
    mountedComponent = shallow(<UserEvents {...props} />);
  }
  return mountedComponent;
};

describe('user history Component', () => {
  beforeEach(() => {});
  wrapper = getComponent();

  it('component successfully rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('testing component will recieve props', () => {
    const nextProps = {
      allUserEvents: {
        fetchedEvents: [events]
      }
    };

    wrapper = getComponent();
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state().isLoading).toEqual(false);
  });

  it('testing load more function', () => {
    wrapper = getComponent();
    wrapper.instance().loadMoreContent();
    expect(wrapper.state().pageNo).toEqual(2);
    expect(wrapper.state().limit).toEqual(6);
  });
  it('testing load more function', () => {
    wrapper = getComponent();
    wrapper.instance().handleClose();
    expect(wrapper.state().showModal).toEqual(false);
  });
  it('testing load more function', () => {
    wrapper = getComponent();
    wrapper.instance().handleShowEditModal();
    expect(wrapper.state().showModal).toEqual(true);
  });
  it('should have a method that handles getMoreEvents', () => {
    expect(wrapper.instance().getMoreEvents).toBeDefined();
  });
  it('should have a method that handles getMoreEvents', () => {
    expect(wrapper.instance().loadMoreContent).toBeDefined();
  });
  it('should have a method that handles getMoreEvents', () => {
    expect(wrapper.instance().handleShowEditModal).toBeDefined();
  });
  it('should have a method that handles getMoreEvents', () => {
    expect(wrapper.instance().handleClose).toBeDefined();
  });
  it('should have a method that handles getMoreEvents', () => {
    expect(wrapper.instance().handleDeleteEvent).toBeDefined();
  });

  describe('Connected Component', () => {
    describe('Connected Events Component', () => {
      it('component successfully rendered', () => {
        const store = mockStore({
          events
        });
        wrapper = shallow(<connect
          store={store}
          allUserEvents={allUserEvents}
          moreEvents={allUserEvents.moreEvents}
          getUsersEvents={getUsersEvents}
          user={user}
          events={events}
          deleteEvent={deleteEvent}
          editEvent={editEvent}
        />);
        expect(wrapper.length).toBe(1);
      });
    });
  });
});
