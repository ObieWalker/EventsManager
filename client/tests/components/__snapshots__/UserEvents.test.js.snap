// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user history Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserEvents
    allUserEvents={
      Object {
        "events": Array [
          Object {
            "event1": Object {
              "Center": Object {
                "address": "23, Iyanla Street",
                "city": "Ikate",
                "image": "",
                "name": "The Sugar Mama",
              },
              "centerId": 3,
              "createdAt": "2018-05-24T16:16:11.648Z",
              "date": "2018-06-04",
              "eventType": "Party",
              "guestNo": 60,
              "id": 3,
              "isCancelled": null,
              "updatedAt": "2018-05-24T16:16:11.648Z",
              "userId": 4,
            },
            "event2": Object {
              "Center": Object {
                "address": "23, Segun Samuel Street",
                "city": "Denrele",
                "image": "",
                "name": "The Hot Abode",
              },
              "centerId": 6,
              "createdAt": "2018-07-24T16:16:11.648Z",
              "date": "2018-9-04",
              "eventType": "Meeting",
              "guestNo": 20,
              "id": 4,
              "isCancelled": null,
              "updatedAt": "2018-07-24T16:16:11.648Z",
              "userId": 5,
            },
            "event3": Object {
              "Center": Object {
                "address": "34, Admiralty way",
                "city": "Lekki",
                "image": "",
                "name": "No Chill",
              },
              "centerId": 9,
              "createdAt": "2018-05-20T16:16:11.648Z",
              "date": "2018-06-09",
              "eventType": "Party",
              "guestNo": 200,
              "id": 9,
              "isCancelled": null,
              "updatedAt": "2018-05-21T16:16:11.648Z",
              "userId": 8,
            },
          },
        ],
        "moreHistory": true,
      }
    }
    deleteEvent={[MockFunction]}
    editEvent={[MockFunction]}
    events={
      Object {
        "event1": Object {
          "Center": Object {
            "address": "23, Iyanla Street",
            "city": "Ikate",
            "image": "",
            "name": "The Sugar Mama",
          },
          "centerId": 3,
          "createdAt": "2018-05-24T16:16:11.648Z",
          "date": "2018-06-04",
          "eventType": "Party",
          "guestNo": 60,
          "id": 3,
          "isCancelled": null,
          "updatedAt": "2018-05-24T16:16:11.648Z",
          "userId": 4,
        },
        "event2": Object {
          "Center": Object {
            "address": "23, Segun Samuel Street",
            "city": "Denrele",
            "image": "",
            "name": "The Hot Abode",
          },
          "centerId": 6,
          "createdAt": "2018-07-24T16:16:11.648Z",
          "date": "2018-9-04",
          "eventType": "Meeting",
          "guestNo": 20,
          "id": 4,
          "isCancelled": null,
          "updatedAt": "2018-07-24T16:16:11.648Z",
          "userId": 5,
        },
        "event3": Object {
          "Center": Object {
            "address": "34, Admiralty way",
            "city": "Lekki",
            "image": "",
            "name": "No Chill",
          },
          "centerId": 9,
          "createdAt": "2018-05-20T16:16:11.648Z",
          "date": "2018-06-09",
          "eventType": "Party",
          "guestNo": 200,
          "id": 9,
          "isCancelled": null,
          "updatedAt": "2018-05-21T16:16:11.648Z",
          "userId": 8,
        },
      }
    }
    getUsersEvents={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
            6,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
    user={
      Object {
        "createdAt": "2018-05-25T18:04:41.552Z",
        "email": "agberodeyforhere@gmail.com",
        "firstName": "Agbero",
        "id": 17,
        "isAdmin": false,
        "lastName": "Idumota",
        "loginTime": "2018-04-26T16:15:47.142Z",
        "password": "parking",
        "updatedAt": "2018-05-25T18:04:41.552Z",
        "username": "agberodeyforhere",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          All Your Upcoming Events.
        </h3>,
        <div>
           
          You have no booked events
          <br />
          <br />
        </div>,
        <Dialog
          autoDetectWindowHeight={true}
          autoScrollBodyContent={true}
          modal={false}
          onRequestClose={[Function]}
          open={false}
          repositionOnUpdate={true}
        >
          <Connect(EditModal)
            event=""
            hideModal={[Function]}
          />
        </Dialog>,
      ],
      "style": Object {
        "margin": "auto",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "All Your Upcoming Events.",
        },
        "ref": null,
        "rendered": "All Your Upcoming Events.",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "You have no booked events",
            <br />,
            <br />,
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "You have no booked events",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoDetectWindowHeight": true,
          "autoScrollBodyContent": true,
          "children": <Connect(EditModal)
            event=""
            hideModal={[Function]}
          />,
          "modal": false,
          "onRequestClose": [Function],
          "open": false,
          "repositionOnUpdate": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "event": "",
            "hideModal": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            All Your Upcoming Events.
          </h3>,
          <div>
             
            You have no booked events
            <br />
            <br />
          </div>,
          <Dialog
            autoDetectWindowHeight={true}
            autoScrollBodyContent={true}
            modal={false}
            onRequestClose={[Function]}
            open={false}
            repositionOnUpdate={true}
          >
            <Connect(EditModal)
              event=""
              hideModal={[Function]}
            />
          </Dialog>,
        ],
        "style": Object {
          "margin": "auto",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "All Your Upcoming Events.",
          },
          "ref": null,
          "rendered": "All Your Upcoming Events.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "You have no booked events",
              <br />,
              <br />,
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "You have no booked events",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoDetectWindowHeight": true,
            "autoScrollBodyContent": true,
            "children": <Connect(EditModal)
              event=""
              hideModal={[Function]}
            />,
            "modal": false,
            "onRequestClose": [Function],
            "open": false,
            "repositionOnUpdate": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "event": "",
              "hideModal": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
